apps/gp2-server/src/data-providers/user.data-provider.ts

```diff
@@ -105,26 +121,44 @@ export class UserSquidexDataProvider implements UserDataProvider {
   }
 }
 
-const mapUserFields = ({
-  role,
-  region,
-  degrees,
-  ...user
-}: gp2.UserUpdateDataObject) => {
-  const mappedDegrees = degrees?.map((degree) => {
+type UserCreateDataObjectEnumFields = Pick<
+  gp2.UserCreateDataObject,
+  'degrees' | 'region' | 'role'
+>;
+type UserCreateInputEnumFields = {
+  degree?: `${UsersDataDegreeEnum}`[];
+  region: UsersDataRegionEnum;
+  role: UsersDataRoleEnum;
+};
+type UserUpdateDataObjectEnumFields = Pick<
+  gp2.UserUpdateDataObject,
+  'degrees' | 'region' | 'role'
+>;
+type UserUpdateInputEnumFields = Partial<UserCreateInputEnumFields>;
+
+function mapUserFields(
+  input: UserCreateDataObjectEnumFields,
+): UserCreateInputEnumFields;
+function mapUserFields(
+  input: UserUpdateDataObjectEnumFields,
+): UserUpdateInputEnumFields;
+function mapUserFields(
+  input: UserCreateDataObjectEnumFields | UserUpdateDataObjectEnumFields,
+): UserCreateInputEnumFields | UserUpdateInputEnumFields {
```

Comment:
I know it's doing the overloading but I found that very difficult to read. 

Also, we've lost the destructuring, is this intentional?